processor/src/debug.rs

```diff
@@ -114,6 +120,44 @@ impl VmStateIterator {
             (None, false)
         }
     }
+
+    //TODO: refactor such that we refactor common logic between `prev()` and `next` into a shared function.
+    pub fn prev(&mut self) -> Option<VmState> {
+        if self.clk == 0 {
+            return None;
+        }
+
+        self.clk -= 1;
+
+        // if we are changing directions we must decrement the clk counter.
+        if self.forward && self.clk > 0 {
+            self.clk -= 1;
+            self.forward = false;
+        }
+
+        let ctx = self.system.get_ctx_at(self.clk);
+
+        let op = if self.clk == 0 {
+            None
+        } else {
+            Some(self.decoder.debug_info().operations()[self.clk as usize - 1])
+        };
+
+        let (asmop, is_start) = self.get_asmop();
+        if is_start {
+            self.asmop_idx -= 1;
+        }
+
+        Some(VmState {
+            clk: self.clk,
+            ctx,
+            op,
+            asmop,
+            fmp: self.system.get_fmp_at(self.clk),
+            stack: self.stack.get_state_at(self.clk),
+            memory: self.chiplets.get_mem_state_at(ctx, self.clk),
+        })
+    }
```

Comment:
Should we not implement [DoubleEndedIterator](https://doc.rust-lang.org/std/iter/trait.DoubleEndedIterator.html) instead on `VmStateIterator`?

Reply:
`DoulbeEndedIterator` has been implemented.